persistence/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/ChildPersistentEntity.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.persistence
+
+import akka.actor.{ ActorContext, ActorRef, PoisonPill }
+import akka.util.Timeout
+import com.lightbend.lagom.internal.scaladsl.persistence.PersistentEntityActor
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+
+/**
+ * A persistent entity that runs as a child actor of the current actor context.
+ *
+ * @param actor The actor for direct access if necessary.
+ */
+sealed class ChildPersistentEntity[Command](val actor: ActorRef) {
+  /**
+   * Send a message to the actor.
+   */
+  def !(msg: Command)(implicit sender: ActorRef): Unit = actor ! msg
+
+  /**
+   * Ask the actor with a command.
+   */
+  def ?[Cmd <: Command with PersistentEntity.ReplyType[_]](command: Cmd)(implicit timeout: Timeout): Future[command.ReplyType] = {
+    import akka.pattern.ask
+
+    (actor ? command).asInstanceOf[Future[command.ReplyType]]
+  }
+
+  /**
+   * Forward the command to the entity.
+   */
+  def forward(msg: Command)(implicit ctx: ActorContext): Unit = actor forward msg
+
+  /**
+   * Stop the persistent entity.
+   */
+  def stop(): Unit = {
+    actor ! PersistentEntityActor.Stop
+  }
+}
+
+object ChildPersistentEntity {
+
+  /**
+   * Instantiate a persistent entity as a child of the current actor context.
+   *
+   * Note that since this is a direct child, this provides no guarantee that this is the only entity with the given id
+   * running in the cluster, or even in the actor system. It is up to the actor that instantiates this to ensure that
+   * only one instance of the entity across the cluster is instantiated (by using cluster sharding to distribute
+   * itself, for example), or to deal with inconsistencies that may arise due to having multiple entities instantiated
+   * in the cluster.
+   *
+   * @param factory   The entity factory.
+   * @param entityId  The entity ID.
+   * @param actorName The actor name.
+   */
+  def apply[P <: PersistentEntity: ClassTag](factory: () => P, entityId: String, actorName: String)(implicit ctx: ActorContext): ChildPersistentEntity[P#Command] = {
+
+    ctx.child(actorName) match {
+
+      case Some(actorRef) => new ChildPersistentEntity(actorRef)
+      case None =>
+
+        val conf = ctx.system.settings.config.getConfig("lagom.persistence")
+        val snapshotAfter: Option[Int] = conf.getString("snapshot-after") match {
+          case "off" => None
+          case _     => Some(conf.getInt("snapshot-after"))
+        }
+
+        val entity = factory()
+
+        val actorRef = ctx.actorOf(PersistentEntityActor.props(
+          persistenceIdPrefix = entity.entityTypeName,
+          entityId = Some(entityId),
+          entityFactory = () => entity,
+          snapshotAfter = snapshotAfter,
+          passivateAfterIdleTimeout = Duration.Undefined
```

Comment:
Do we want to ignore passivation timeout settings here?

Reply:
Yes - control of the lifecycle of the entity belongs with the parent, not with the entity itself.

Note that when a persistent entity passivates itself, it sends a `ShardRegion.Passivate` message to its parent, assuming that its parent is a shard region. This causes the shard region to start buffering all messages to the entity, and sends the entity a `PersistentEntityActor.Stop` message, which tells the entity its now safe to start shutting down, its not going to receive any more messages. This is a contract between the shard region and the persistent entity actor. If the parent of the persistent entity actor is not a shard region (which it isn't if using this API), then no such contract exists and it makes no sense for the entity to be sending its parent a `ShardRegion.Passivate` message.

If the user wants to implement passivation, they can do that on their own parent actor, where passivation logic belongs.

Reply:
Fair enough, in that case I think we should document how passivation is disabled for child persistent entities (but `snapshot-after` is reused). Users may have to manually implement some bizz-specific passivation mechanism to avoid OOME. There are docs for other areas were users will have to manually workout features provided by Lagom in _regular_ persistent entities (e.g. uniqueness, graceful shutdown...).